java/flight/flight-core/src/test/java/org/apache/arrow/flight/TestErrorMetadata.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import org.apache.arrow.flight.perf.impl.PerfOuterClass;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.protobuf.Any;
+import com.google.rpc.Status;
+
+import io.grpc.Metadata;
+import io.grpc.StatusRuntimeException;
+import io.grpc.protobuf.StatusProto;
+import io.grpc.protobuf.lite.ProtoLiteUtils;
+
+public class TestErrorMetadata {
+  private static final Metadata.BinaryMarshaller<PerfOuterClass.Perf> marshaller =
+      ProtoLiteUtils.metadataMarshaller(PerfOuterClass.Perf.getDefaultInstance());
+
+  @Test
+  public void testMetadata() {
+    PerfOuterClass.Perf perf = PerfOuterClass.Perf.newBuilder()
+                .setStreamCount(12)
+                .setRecordsPerBatch(1000)
+                .setRecordsPerStream(1000000L)
+                .build();
+    try (final BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+        final FlightServer s =
+             FlightTestUtil.getStartedServer(
+               (location) -> FlightServer.builder(allocator, location, new TestFlightProducer(perf)).build());
+          final FlightClient client = FlightClient.builder(allocator, s.getLocation()).build()) {
+      client.getStream(new Ticket("abs".getBytes()));
+    } catch (FlightRuntimeException fre) {
+      FlightMetadata metadata = fre.status().metadata();
+      Assert.assertNotNull(metadata);
+      Assert.assertEquals(1, metadata.size());
+      Assert.assertTrue(metadata.containsKey("Testing 1 2 3"));
+      Assert.assertEquals(perf, metadata.get("Testing 1 2 3", marshaller::parseBytes));
+    } catch (Exception e) {
```

Comment:
You might find it easier to just declare the test `throws Exception`.

Reply:
:+1: